python/graphstorm/wholegraph/wholegraph.py

```diff
@@ -457,16 +472,88 @@ def load_from_file(
             The prefix of the file.
         num_files : int
             The number of files to load.
-
+        wg_optimizer : WholeMemoryOptimizer or None
+            The WholeGraph sparse optimizer to be attached to the WholeGraph embedding.
         Returns
         -------
         None
         """
+
+        if self._tensor is None and self._use_wg_optimizer:
+            self.attach_wg_optimizer(wg_optimizer)
```

Comment:
Shall we assert wg_optimizer is not None here?

Reply:
Good point. I am thinking of two scenarios:
(1) user specifies wg_optimizer when calling load_from_file:
    In this sense, we need to replace the existing optimizer/tensor/module by a new one before reading from files. So we need to call attach-wg_optimizer again
(2) user does not specify wg_optimizer
    In this sense, by default, we assume user has self.optimizer defined (if use_wg_optimizer is on) and all we need to do is update the _tensor with the one loaded from files.

Reply:
Can you put the description of these two scenarios in the function docstring?
Actually there are three scenarios:
1) learnable embedding with wg_optimizer
2) learnable embedding with wg_optimizer = None
3) regular embedding.